### PR TITLE
CATROID-1131 Refactor BrickCategoryAdapter and Fragment

### PR DESCRIPTION
**Pull request by Walcho1125**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*
https://jira.catrob.at/browse/CATROID-1131
Refactor BrickCategoryAdapter and BrickCategoryAdapter to Kotlin

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by dandrawes — APPROVED** [review]
Well done!

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/BrickCategoryFragment.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        menu.findItem(R.id.delete).isVisible = false
+        menu.findItem(R.id.copy).isVisible = false
+        menu.findItem(R.id.backpack).isVisible = false
+        menu.findItem(R.id.comment_in_out).isVisible = false
+        menu.findItem(R.id.catblocks).isVisible = false
+        menu.findItem(R.id.catblocks_reorder_scripts).isVisible = false
+        menu.findItem(R.id.find).isVisible = false
+    }
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/BrickCategoryFragment.kt:135]:
```suggestion

```

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/BrickCategoryFragment.kt [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val isRestoringPreviouslyDestroyedActivity = savedInstanceState != null
+        if (isRestoringPreviouslyDestroyedActivity) {
+            parentFragmentManager.popBackStack(BRICK_CATEGORY_FRAGMENT_TAG, FragmentManager.POP_BACK_STACK_INCLUSIVE)
+            return
+        }
+        setHasOptionsMenu(true)
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/BrickCategoryFragment.kt:79]:
```suggestion
        setHasOptionsMenu(false)
```
Please check if this works, the other solution is a bit odd.

**Review by gPathpp — CHANGES_REQUESTED** [review]

**Review by gPathpp — APPROVED** [review]
As discussed, the main refactoring/implementation happens in another (GSoC) ticket.

**Pull request merged**